### PR TITLE
luci-app-frpc/frps:add service status display

### PR DESCRIPTION
**Pull request by msylgj**
reference from luci-app-smartdns
Special thanks to Nick Peng <pymumu@gmail.com>

Signed-off-by: SiYao Mo <msylgj@vip.qq.com>

### CODE REVIEW

File: applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js [hunk 1]
```diff
@@ -115,12 +116,68 @@ function defOpts(s, opts, params) {
 	}
 }
 
+var conf = 'frpc';
```
Comment (feckert) [inline applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js:119]: I would not move this into its own var.
From my point of view, that makes no sense.
This is not changed at runtime.

File: applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js [hunk 2]
```diff
@@ -115,12 +116,68 @@ function defOpts(s, opts, params) {
...
+		getPid()
+	]);
+}
+
+function renderStatus(res) {
+	var renderHTML = "";
+	var isRunning = res[0];
+
+	if (isRunning) {
+		renderHTML += "<span style=\"color:green;font-weight:bold\">&nbsp;&nbsp;&nbsp;&nbsp;" + _("frp Client") + " - " + _("RUNNING") + "</span>";
```
Comment (feckert) [inline applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js:148]: Could we not use `String.format` there?
That would be clearer

File: applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js [hunk 3]
```diff
@@ -115,12 +116,68 @@ function defOpts(s, opts, params) {
...
+	} else {
+		renderHTML += "<span style=\"color:red;font-weight:bold\">&nbsp;&nbsp;&nbsp;&nbsp;" + _("frp Client") + " - "  + _("NOT RUNNING") + "</span>";
+		return renderHTML;
```
Comment (feckert) [inline applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js:151]: Not needed. Only one return statement at function end

**Review by feckert — COMMENTED** [review]
The same comments also apply to the `frps.js `file

File: applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js (hunk 1 above)
Comment (msylgj) [reply applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js:119]: yeap, just make functions more versatile. I'll remove it.

File: applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js (hunk 3 above)
Comment (msylgj) [reply applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js:151]: a mistake.

File: applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js (hunk 2 above)
Comment (msylgj) [reply applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js:148]: shell i just remove those '&nbsp;'?

**Comment by msylgj** [comment]
@feckert plz CR again.

File: applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js (hunk 2 above)
Comment (feckert) [reply applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js]: That is what I have meant:
`String.format("<span style=\"color:green;font-weight:bold\">&nbsp;&nbsp;&nbsp;&nbsp;%s-%s</span>", _("frp Client") _("RUNNING"));`

Comment (msylgj) [reply applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js]: wow, get it.

Comment (feckert) [reply applications/luci-app-frpc/htdocs/luci-static/resources/view/frpc.js]: looks better :+1:

**Comment by feckert** [comment]
Looks good. can you then merge the commits so that we only have one?

**Comment by msylgj** [comment]
> Looks good. can you then merge the commits so that we only have one?

of course! I will do it in a few hours.

**Comment by msylgj** [comment]
> > Looks good. can you then merge the commits so that we only have one?
> 
> of course! I will do it in a few hours.

@feckert I've done it. and also a pr same of this to the branch openwrt-21.02. https://github.com/openwrt/luci/pull/5145

**Comment by feckert** [comment]
Thank you for your contribution

**Comment by msylgj** [comment]
> Thank you for your contribution

my pleasure.

**Pull request merged**